lamindb/curators/__init__.py

```diff
@@ -173,8 +173,16 @@ class Curator:
     """
 
     def __init__(self, dataset: Any, schema: Schema | None = None):
-        self._dataset: Any = dataset  # pass the dataset as a UPathStr or data object
         self._artifact: Artifact = None  # pass the dataset as a non-curated artifact
+        self._dataset: Any = dataset  # pass the dataset as a UPathStr or data object
+        if isinstance(dataset, Artifact):
+            self._artifact = dataset
+            # tiledbsoma logic is dealt in TiledbsomaCatCurator
+            if (
+                self._artifact.otype != "tiledbsoma"
+                or self._artifact.suffix != ".tiledbsoma"
+            ):
```

Comment:
Maybe this should actually check that this is `anndata` or `dataframe` in addition to checking that it is `tiledbsoma`?

Reply:
I agree, but I think that's even harder, isn't it? For example, how to check if an artifact is anndata? `otype="anndata"` or `suffix=".h5ad"` or `suffix=".h5"? dataframe also has different suffixes.

Reply:
I think that this `otype` checking should _solely_ be done based on the `otype`, not taking into account `.suffix`.

Assuring integrity of the artifact state is another concern that we can debate maintaining in another function, but not with custom "`or`" logic that's only going to bite us with unexpected behavior and hard-to-maintain code.

Reply:
The danger is old artifacts might not have `otype` properly populated. Also for `.csv` files?

Reply:
An `or` check is unpredictable. If `otype` isn't populated this should throw an error.

The `Curator` is a very complicated thing; you shouldn't be able to feed in a `.csv` file. Only artifacts with `otype` `DataFrame`.

We have to get much more rigorous with enforcing `otype`-integrity across the codebase. So if you keep a `.suffix` check here it'll have to be refactored sooner or later.

Reply:
(The `.suffix` is a wild-west filesystem thing to a large degree; the `otype` we can control and validate.)